src/kernel/src/hal/platform/hyperlight/mod.rs

```diff
@@ -226,22 +227,68 @@ pub fn shutdown(_status: usize) -> ! {
 ///
 /// A new boot information structure.
 ///
-pub fn parse_bootinfo(magic: u32, info: usize) -> Result<BootInfo, Error> {
-    // Check if magic number matches what we expect.
-    if magic != ::config::hyperlight::DEFAULT_BOOT_MAGIC {
-        let reason: &str = "invalid boot magic number";
-        error!("parse_bootinfo(): magic={:#010x}, info={:#010x} (error={})", magic, info, reason);
-        return Err(Error::new(ErrorCode::InvalidArgument, reason));
+pub fn parse_bootinfo(_: u32, _: usize) -> Result<BootInfo, Error> {
+    trace!("parse_bootinfo()");
+
+    extern "C" {
+        static __KERNEL_END: u8;
     }
 
-    trace!("parse_bootinfo(): magic={:#010x}, info={:#010x}", magic, info);
+    let peb_base =
+        unsafe { ::sys::mm::align_up(&__KERNEL_END as *const u8 as usize, PAGE_ALIGNMENT) };
+
+    unsafe {
+        ProcessEnvironmentBlock::init(peb_base as *mut HyperlightPEB)?;
+        ProcessEnvironmentBlock::set_guest_function_dispatch_ptr(0xdeadbeef)?;
```

Comment:
Could you add a comment why this is required? Better yet, we can have an issue for this and link it here.

Reply:
Hyperlight has a hard requirement on the dispatch function address not being null: https://github.com/hyperlight-dev/hyperlight/blob/6ab6b2cf496210165b8b67cc446f4b0f4a20e52f/src/hyperlight_host/src/sandbox/uninitialized_evolve.rs#L114-L116